website/components/LoadModal.tsx

```diff
@@ -0,0 +1,162 @@
+      cellId: "1",
+      parentImageUrl: "",
+      parentImageDownloadHref: "",
+      // Enable first three channels by default
+      viewerChannelSettings: {
+        groups: [
+          {
+            name: "Channels",
+            channels: [
+              { match: [0, 1, 2], enabled: true },
+              { match: "(.+)", enabled: false },
+            ],
+          },
+        ],
+      },
+    };
+    props.onLoad(appProps);
+    addRecentDataUrl({ url: urlInput, label: urlInput });
+    setShowModal(false);
+  };
+
+  // Set up fuse for fuzzy searching on the labels of recent datasets
+  const fuse = useMemo(() => {
+    return new Fuse(recentDataUrls, {
+      keys: ["label"],
+      isCaseSensitive: false,
+      shouldSort: true, // sorts by match score
+      ignoreLocation: true, // search more than first 60 characters
+      findAllMatches: true, // return all matches
+    });
+  }, [recentDataUrls]);
+
+  // This search could be done using a transition if needed, but since there is a max of 100 urls,
+  // performance hits should be minimal.
+  const autoCompleteOptions: { label: React.ReactNode; value: string }[] = useMemo(() => {
+    let filteredItems: RecentDataUrl[] = [];
+    if (urlInput === "") {
+      // Show first 20 recent data urls
+      filteredItems = recentDataUrls.slice(0, MAX_RECENT_URLS_TO_DISPLAY);
+    } else {
+      // Show first 20 search results
+      filteredItems = fuse
+        .search(urlInput)
+        .slice(0, MAX_RECENT_URLS_TO_DISPLAY)
+        .map((option) => option.item);
+    }
+    return filteredItems.map((item) => {
+      return {
+        label: <MiddleTruncatedText text={item.label} />,
+        value: item.url,
+      };
+    });
+  }, [urlInput, fuse]);
+
+  const getAutoCompletePopupContainer = modalContainerRef.current ? () => modalContainerRef.current! : undefined;
+
+  return (
+    <ModalContainer ref={modalContainerRef}>
+      <Button type="link" onClick={() => setShowModal(!showModal)}>
+        <UploadOutlined />
+        Load
+      </Button>
+      <Modal
+        open={showModal}
+        title={"Load"}
+        onCancel={() => {
+          setShowModal(false);
+        }}
+        getContainer={modalContainerRef.current || undefined}
+        okButtonProps={{}}
+        footer={
+          <Button type="default" onClick={() => setShowModal(false)}>
+            Cancel
+          </Button>
+        }
+        destroyOnClose={true}
+      >
+        <p style={{ fontSize: "16px" }}>Provide the URL to load your OME-Zarr or OME-TIFF* data.</p>
+        <p style={{ fontSize: "12px" }}>
+          <i>*Note: this tool is intended for OME-Zarr use. Large {"(> 100 MB)"} OME-TIFF files are not supported.</i>
+        </p>
+        <FlexRow $gap={6}>
+          <AutoComplete
+            value={urlInput}
+            onChange={(value) => setUrlInput(value)}
+            onSelect={(value) => {
+              setUrlInput(value as string);
+            }}
```

Comment:
another minor nit: may be able to use `setUrlInput` without the closure here, i.e. `onChange={setUrlInput}`